### PR TITLE
Fixes #81 Feature request: Simulate to steady-state

### LINKED ISSUE #81: Feature request: Simulate to steady-state
Especially for QSP models with endogenous production of compounds, it is often necessary to start the simulation with initial state other then 0 for the most molecules. To determine this initial state, the model can be simulated for a long time and the end state set to the initial state. A simplified workflow looks like this:

1) Simulate the model with default initial state for long time
2) Iterate through all outputs and the speciesInitialValues to the end value of the output
3) Optionally, save end result to an excel file to be able to load it in MoBi.

There are some questions:
1) MoBi-import requires the column "Neg. values allowed" for every entry. How do I get this information from within R?
2) Initial values given by an equation should not be overwritten. However, getSpeciesInitialValue() returns a wrong value for "IsFormula" (#60). How to overcome this limitation?
3) What are the special cases that should be ignored or treated in a different way?
3a) How to treat the EndogenousIgG compartment?
3b) How to treat saliva values?

Anything else?

### PR DESCRIPTION
**Pull request by PavelBal**
Get the steady-state values of species. The steady-state is considered to be the last value of the simulation with sufficiently long simulation time, i.e., where the rates of the processes do not change.

Returns a list containing two data frames - moleculesSteadyState and parametersSteadyState. The data frame moleculesSteadyState contains columns "Container Path", "Molecule Name", "Is Present", "Value", "Units", "Scale Divisor", "Neg. Values Allowed".

The data frame parametersSteadyState. contains columns "Container Path", "Parameter Name", "Value", "Units".

All species with the unit "µmol" that are located within the "Organism"-structure are treated as molecules and stored in the moleculesSteadyState-data frame. That means that local state variable parameters with the dimension "Amount" are treated as molecules.

All species with units other then "µmol" OR the top container being not "Organism" are stored in the parametersSteadyState-data frame.

Example: the local state variable parameter "Organism|Fat|Intracellular|myReaction|ParamName" will be stored in the moleculesSteadyState-data frame.

The global parameter "myReaction|ParamName" will be stored in the parametersSteadyState-data frame.
The Container Path does not include the name of the simulation.

If no entries matching the criteria (speciesNames and ignoreIfFormula) could be generated, an emtpy data frame is returned (length = zero).

### CODE REVIEW

**Review by sfrechen — APPROVED** [review]

**Comment by PavelBal** [comment]
Please review and, if applicable, merge.

File: src/code/getSpeciesSteadyState.R [hunk 1]
```diff
@@ -0,0 +1,119 @@
+
+#' Get the steady-state values of species. The steady-state is considered to be the last value of the simulation with sufficiently
+#' long simulation time, i.e., where the rates of the processes do not change.
+#'
+#' @param speciesNames Vector of species names which steady-state values will be returned. By default, values of all species are returned. This also includes
+#' values of "state variable" parameters, as these are treated like species.
+#' @param steadyStateTime Simulation time (minites). Must be long enough for system to reach a steady-state. Default is 1000 minutes.
```
Comment (Yuri05) [inline src/code/getSpeciesSteadyState.R]: I would not define any **default** for the simulation time. It must be in the responsibility of a user to set the time properly.

File: src/code/getSpeciesSteadyState.R [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+#' @param ignoreIfFormula If "TRUE", species with initial values defined by formula are not included. Default is "TRUE".
+#' @param DCI_Info The DCI Info structure containing the DCI handle and all settings.
+#'
+#' @return A list containing two data frames - moleculesSteadyState and parametersSteadyState. The data frame moleculesSteadyState contains columns
+#' "Container Path", "Molecule Name", "Is Present", "Value", "Units", "Scale Divisor", "Neg. Values Allowed".
+#' The data frame parametersSteadyState. contains columns
+#' "Container Path", "Parameter Name", "Value", "Units".
+#' All species with the unit "�mol" that are located within the "Organism"-structure are treated as molecules and stored in the moleculesSteadyState-data frame.
+#' That means that local state variable parameters with the dimension "Amount" are treated as molecules.
+#' All species with units other then "�mol" OR the top container being not "Organism" are stored in the parametersSteadyState-data frame.
```
Comment (Yuri05) [inline src/code/getSpeciesSteadyState.R]: >All species with the unit "µmol" that are located within the "Organism"-structure are treated as molecules 
>All species with units other then "µmol" OR the top container being not "Organism" are stored in the parametersSteadyState...

Are those reasonable assumptions? E.g. there are - even in "standard" models - molecule amounts which are not located under Organism (e.g. solid/insoluble drug located under Applications)

File: src/code/getSpeciesSteadyState.R (hunk 1 above)
Comment (PavelBal) [reply src/code/getSpeciesSteadyState.R]: Agree.

File: src/code/getSpeciesSteadyState.R (hunk 2 above)
Comment (PavelBal) [reply src/code/getSpeciesSteadyState.R]: It is indeed not a perfect solution, but I do not know of any other way to distinguish between species and state-variable parameters.

When working from the toolbox, it does not make any difference - all entries returned by "getSpeciesSteadyState()" will be species. The difference between "Molecule" and "Parameter" is only important when exporting the steady-state to xls and importing in MoBi.
Special cases, when molecules are located outside of "Organism", must be treated by the user manually - and the documentation describes it clearly.

As far as I know, amounts of molecules in Applications do not really matter, so it OK not to treat them properly.

But, yes, if there is another way to differentiate between "Molecule" and "Parameter", it should be used.

Comment (Yuri05) [reply src/code/getSpeciesSteadyState.R]: OK, just checked it. You are right, currently there is no another possibility.
State variable parameters are defined as variables already in the simulation XML

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**